app/src/main/java/com/example/tasky/MainActivity.kt

```diff
@@ -32,11 +44,40 @@ class MainActivity : ComponentActivity() {
         }
         enableEdgeToEdge()
         setContent {
+            val navController = rememberNavController()
+
+            val startDestination: Any = if (isLoggedInState.value) {
+                LoginNav
+            } else RegisterNav
+
             TaskyTheme {
-                if (isLoggedInState.value) {
-                    LoginScreenContent()
-                } else RegisterAccountContent()
+                NavHost(navController, startDestination = startDestination) {
+                    composable<RegisterNav> {
+                        RegisterAccountContent(navController = navController)
+                    }
+                    composable<LoginNav> {
+                        LoginScreenContent(navController = navController)
+                    }
+                    composable<CalendarNav> {
+                        // todo create Composable
+                        Box(
+                            modifier = Modifier
+                                .fillMaxSize()
+                                .background(Color.Green)
+                        )
+                    }
+                    // todo add other screens
+                }
```

Comment:
I like to work with one navigation graph per feature, since then you can conveniently pop the whole graph from the back stack when logging in for example without risking there are some screens left on the back stack.

Works just like defining routes:
```
@Serializable
data object AuthGraph

NavHost(...) {
    navigation<AuthGraph> {
        composable<LoginRoute> {
        
        }
        ...
    }
}
```